packages/sdk/src/vdp/artifact/ArtifactClient.ts

```diff
@@ -0,0 +1,249 @@
+
+      if (data.nextPageToken) {
+        files.push(
+          ...(await this.listKnowledgeBaseFiles({
+            ownerId,
+            kbId,
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination: false,
+          }))
+        );
+      }
+
+      return Promise.resolve(files);
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  }
+
+  async uploadKnowledgeBaseFile(props: UploadKnowledgeBaseFileRequest): Promise<File> {
+    const { ownerId, kbId, payload } = props;
+    try {
+      const data = await this._client.post<{ file: File }>(`/namespaces/${ownerId}/knowledge-bases/${kbId}/files`, {
+        body: JSON.stringify(payload),
+      });
+      return Promise.resolve(data.file);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async deleteKnowledgeBaseFile({ fileUid }: DeleteKnowledgeBaseFileRequest): Promise<void> {
+    try {
+      await this._client.delete(`/knowledge-bases/files?fileUid=${fileUid}`);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listChunks(props: ListChunksRequest & { enablePagination: true }): Promise<ListChunksResponse>;
+  async listChunks(props: ListChunksRequest & { enablePagination: false }): Promise<Chunk[]>;
+  async listChunks(props: ListChunksRequest & { enablePagination: boolean }): Promise<ListChunksResponse | Chunk[]> {
+    const { ownerId, kbId, fileUid, pageSize, pageToken, enablePagination } = props;
+
+    try {
+      const chunks: Chunk[] = [];
+
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${ownerId}/knowledge-bases/${kbId}/chunks`,
+        pageSize,
+        pageToken,
+        fileUid,
+      });
+
+      const data = await this._client.get<ListChunksResponse>(queryString);
+
+      if (enablePagination) {
+        return Promise.resolve(data);
+      }
+
+      chunks.push(...data.chunks);
+
+      if (data.nextPageToken) {
+        chunks.push(
+          ...(await this.listChunks({
+            ownerId,
+            kbId,
+            fileUid,
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination: false,
+          }))
+        );
+      }
+
+      return Promise.resolve(chunks);
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  }
+
+  async updateChunk(props: UpdateChunkRequest): Promise<Chunk> {
+    const { chunkUid, retrievable } = props;
+    try {
+      const data = await this._client.post<{ chunk: Chunk }>(`/chunks/${chunkUid}`, {
+        body: JSON.stringify({ retrievable }),
+      });
+      return Promise.resolve(data.chunk);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async getChunkContent({ ownerId, kbId, chunkUid }: GetChunkContentRequest): Promise<string> {
+    try {
+      const data = await this._client.get<{ content: string }>(`/namespaces/${ownerId}/knowledge-bases/${kbId}/chunks/${chunkUid}/content`);
```

Comment:
```suggestion
      const data = await this._client.get<GetChunkContentResponse>(`/namespaces/${ownerId}/knowledge-bases/${kbId}/chunks/${chunkUid}/content`);
```

We need to export every type related to request and response so use can utilitze them. Please add them in for all remaining query